docs/content/en/functions/images/index.md

```diff
@@ -51,14 +51,12 @@ images.Text TEXT DICT)
 The following example will add the text `Hugo rocks!` to the image with the specified font, color, size and position.
 
 ```
-{{ $font := resources.Get "/fonts/font.ttf" }}
```

Comment:
I may have given conflicting feedback re font -- and we can take that in its own separate feature -- but I now think this would be really useful. But see my comment re. hash key.

Reply:
Do you want to include font support for this PR or work that feature out in a separate PR?

If we implement fonts, we need to make sure that the pipeline and font is cached between build/restart. Using resources.Get we utilize the existing logic, for local and remote fonts, including the resource cache. I'm not 100% sure about the pipeline cache, during testing, I had to make changes to the pipeline (even after a restart) to see the changes, but I can imagine that the resource pointer is unpredictable.

I also thought about passing the resource location instead and call the resources get internally but that would cause an import loop.

Reply:
On the hash key, I suppose you refer to [image.go#L237](https://github.com/gohugoio/hugo/blob/master/resources/image.go#L237)?

What do you mean with:
> Replace the font in options with font.Key (there is an interface)

If we want to use resources.Get to support local and remote fonts, I need to pass the resource at some point.

Reply:
There is a more complex story to this, but the short story is:

* If you do what you did and pass the `Resource` (as in: what you get in `resources.Get` etc.) as an option, the caching is in place.
* But, and this is important, you need to make sure that what you create in `HashString` is stable -- which is why I made the comment about:

1. First pull the font from the map.
2. Then replace `font` in the map with `fontResource.Key`.
3. Then use that map to create the hash.

You can do the above in its own PR if you want, or just apply that change to this, if that is most effective.